x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -225,6 +231,18 @@ void processResult(AutodetectResult result) {
         CategoryDefinition categoryDefinition = result.getCategoryDefinition();
         if (categoryDefinition != null) {
             persister.persistCategoryDefinition(categoryDefinition, this::isAlive);
```

Comment:
Now that all this has become more complex than a single call to the persister, I'd be tempted to extract this in a `handleCategoryDefinition` method.